### PR TITLE
Add function for calling update allocation with TWAP.

### PR DESCRIPTION
**Pull request by bweick**

### CODE REVIEW

File: src/api/SocialTradingAPI.ts [hunk 1]
```diff
@@ -935,6 +985,26 @@ export class SocialTradingAPI {
...
 
+  private async assertUpdateAllocationWithTWAP(
+    manager: Address,
+    tradingPool: Address,
+    newAllocation: BigNumber,
+    chunkSize: BigNumber,
+    txOpts: Tx,
+  ): Promise <void> {
+    const poolInfo = await this.fetchNewTradingPoolV2DetailsAsync(tradingPool);
+    // console.log(poolInfo);
```
Comment (controtie) [inline src/api/SocialTradingAPI.ts:996]: remove

File: test/integration/api/SocialTradingAPI.spec.ts [hunk 2]
```diff
@@ -972,6 +974,236 @@ describe('SocialTradingAPI', () => {
...
+        subjectChunkSize = ether(100);
+      });
+
+      test('throws', async () => {
+        return expect(subject()).to.be.rejectedWith(
+          `Passed chunk size must be between ${assetPairBounds[0].bounds.lower} and` +
+          ` ${assetPairBounds[0].bounds.upper}.`
+        );
+      });
+    });
```
Comment (controtie) [inline test/integration/api/SocialTradingAPI.spec.ts:1175]: Not sure how `chunkAuctionPeriod` factors into this. I'll throw out having a test case for invalid `chunkAuctionPeriod` values

Comment (bweick) [reply test/integration/api/SocialTradingAPI.spec.ts:1175]: Suppose we could test to make sure it's greater than 0

Comment (bweick) [reply test/integration/api/SocialTradingAPI.spec.ts:1175]: I guess it depends how thorough we want to make our checks here. Unfortunately there's a lot of calls to the chain that need to be made in order to do some of these checks

Comment (controtie) [reply test/integration/api/SocialTradingAPI.spec.ts:1175]: Sure you guys know better than I, just throwing out a suggestion

File: test/integration/api/SocialTradingAPI.spec.ts [hunk 3]
```diff
@@ -972,6 +974,248 @@ describe('SocialTradingAPI', () => {
...
+      const { timestamp } = await web3.eth.getBlock('latest');
+      const lastRebalanceTimestamp = new BigNumber(timestamp);
+      const entryFee = ether(.01);
+      const profitFeePeriod = ONE_DAY_IN_SECONDS.mul(30);
+      const highWatermarkResetPeriod = ONE_DAY_IN_SECONDS.mul(365);
+      const profitFee = ether(.2);
+      const streamingFee = ether(.02);
+
+      const txHash = await socialTradingAPI.createTradingPoolV2Async(
+        manager,
```
Comment (felix2feng) [inline test/integration/api/SocialTradingAPI.spec.ts:1008]: Damn, that's a lot of arguments

File: test/integration/api/SocialTradingAPI.spec.ts [hunk 4]
```diff
@@ -972,6 +974,248 @@ describe('SocialTradingAPI', () => {
...
+          .format('dddd, MMMM Do YYYY, h:mm:ss a');
+        return expect(subject()).to.be.rejectedWith(
+          `Attempting to rebalance too soon. Rebalancing next ` +
+          `available on ${nextRebalanceFormattedDate}`
+        );
+      });
+    });
+  });
+
   describe('acutalizeFeeAsync', async () => {
```
Comment (felix2feng) [inline test/integration/api/SocialTradingAPI.spec.ts:1219]: actualize

File: test/integration/wrappers/strategies/SocialAllocatorWrapper.spec.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
+/*
+  Copyright 2018 Set Labs Inc.
```
Comment (felix2feng) [inline test/integration/wrappers/strategies/SocialAllocatorWrapper.spec.ts:2]: 2020

**Pull request merged**